### ISSUE TITLE
Multi task Training global step mistake

### ISSUE THREAD

**Issue by CarinaXZZ** [UKPLab/sentence-transformers#2835]
Hi,
I tried to use SentenceTransformerTrainer for multitask learning.
However, when I run
```
trainer = SentenceTransformerTrainer(
    model=model,
    args=training_args,
    train_dataset={'contrastive':train_dataset,
                   'softmax':train_dataset},
    loss={'contrastive':train_loss,
          'softmax':train_loss_cls},
)
```

However, I notcied the global steps as four times of training with just one dataset as follows.
```
trainer = SentenceTransformerTrainer(
    model=model,
    args=training_args,
    train_dataset=train_dataset,
    loss=train_loss,
)
```

Since I used the same dataset for different loss functions, I would expect the training steps will be doubled from the single task training. I wonder if anything goes wrong here.
